api/v1alpha3/owner_labels.go

```diff
@@ -239,6 +239,33 @@ func deleteChildrenSingle(ctx context.Context, c client.Client, childObjectList
 	return deleteRetry.withObjectList(objectList), g.Wait()
 }
 
+// DeleteChildrenWithLabelsParallel deletes all the children of a parent with the resource types defined
+// in a list of ObjectList types and the labels defined in matchingLabels. Deletion will start on
+// all child types in parallel.
+func DeleteChildrenWithLabelsParallel(ctx context.Context, c client.Client, childObjectLists []ObjectList, parent metav1.Object, matchingLabels client.MatchingLabels) (DeleteStatus, error) {
+	for label, value := range MatchingOwner(parent) {
+		matchingLabels[label] = value
+	}
+
+	complete := true
+	for _, childObjectList := range childObjectLists {
+		deleteStatus, err := deleteChildrenSingle(ctx, c, childObjectList, parent, matchingLabels)
```

Comment:
Perhaps I just need to look at the sos PR, but how is parallelism achieved here? Doesn't this block?

Reply:
It doesn't block. It marks all the children for deletion, and then returns a value to indicate that there are still child resources present. Rather than wait for a return value that says no more of that child type are present, we immediately move on to marking the next client type for deletion.

Reply:
Thanks. Makes way more sense then.